Fix for NumAdaptivityCycles postprocessor when manually turning off mesh adaptivity

jessecarterMOOSE opened the pull request:
Slightly modifies the mesh adaptivity logic so that `FEProblemBase::adaptMesh()` always gets run, which allows the adaptivity cycle counter to be reset, allowing `FEProblemBase::getNumCyclesCompleted()` to return to correct value if mesh adaptivity gets shut off manually (ie, calling `Adaptivity::setAdaptivityOn(false)`).

I removed what seemed like a redundant mesh adaptivity check, which was occurring both before `FEProblemBase::adaptMesh()` was called and in the first line of it (different methods were being called but they both require `mesh_refinement_on` to be true). I just removed the first check so the cycle counter would always get reset, and then the `adaptMesh()` function could be skipped from within if necessary.

I also added a couple tests to demonstrate the functionality.

Closes #9746

jessecarterMOOSE committed:
Allow ToggleMeshAdaptivity UO to act after some specified number of steps, add test

jessecarterMOOSE committed:
Modify adaptivity cycles test to run more steps

jessecarterMOOSE committed:
Remove redundant mesh adaptivity check

jessecarterMOOSE committed:
Move adaptivity cycle counter up so it always gets reset

jessecarterMOOSE committed:
Add tests for counting adaptivity cycles when adaptivity is manually shut off, closes #9746

moosebuild commented:
Job [Precheck](https://www.moosebuild.org/job/111871/) on c48df3e wanted to post the following:

Your code requires style changes.
In the top level of your repository, you can run:

`git clang-format 043a0df33293df1b03d33311d3389ce19729f5e4`

Alternatively, apply the patch found [here](http://mooseframework.org/docs/PRs/9748/style.patch).
You can directly apply the patch by running:

`curl -s http://mooseframework.org/docs/PRs/9748/style.patch | git apply -v`

moosebuild commented:
Job [Documentation](https://www.moosebuild.org/job/111887/) on a530286 wanted to post the following:

View the site [here](http://mooseframework.org/docs/PRs/9748/site)

jessecarterMOOSE commented:
Hmm, I fixed one of my commits, squashed it in with the formatting fix, and did a force push. Now some of CIVET tests aren't running. What should I do?

rwcarlsen commented on test/src/userobjects/ToggleMeshAdaptivity.C:
```diff
@@ -52,5 +58,6 @@ ToggleMeshAdaptivity::initialSetup()
 ToggleMeshAdaptivity::execute()
 {
-  checkState();
+  if (_t_step > _steps_to_wait)
```
Should be ``>=`` to allow waiting for zero steps.

permcody replied:
Yeah probably, It's just a test object though so no real impact of choosing one method over another.

rwcarlsen reviewed (commented):
Looks good. @permcody or @jwpeterson probably should give the final okay on the adaptivity logic mod.
